processing/src/main/java/io/druid/segment/data/Offset.java

```diff
@@ -40,6 +40,11 @@
   @CalledFromHotLoop
   public abstract boolean withinBounds();
 
+  /**
+   * Resets the Offset to the position it was created or cloned with.
```

Comment:
Reset is not thread safe, suggest adding in explicit callout to thread safety either in the class description or at least in this method.